src/main/java/Duke.java

```diff
@@ -1,10 +1,212 @@
+            break;
+        case "event":
+            executeEvent(input);
+            break;
+        }
+    }
+
+    public static void executeMark() {
+        System.out.println("Which task would you like to mark as done?");
+        listTasks();
+        int index = getInteger() - 1;
+        tasks[index].markAsDone();
+        System.out.println("Congrats! You've completed:\n" + tasks[index]);
+    }
+
+    public static void executeUnmark() {
+        System.out.println("Which task would you like to unmark as done?");
+        listTasks();
+        int index = getInteger() - 1;
+        tasks[index].markAsNotDone();
+        System.out.println("Aw, you've marked this as undone:\n" + tasks[index]);
+    }
+
+    public static void executeTodo(String input) {
+        String description = getDescription(input);
+        Todo t = new Todo(description);
+        tasks[numTasks] = t;
+        numTasks += 1;
+    }
+
+    public static void executeDeadline(String input) {
+        String[] parsedCommand = parseDeadlineOrEvent(input);
+        String description = parsedCommand[0];
+        String by = parsedCommand[1];
+        Deadline t = new Deadline(description, by);
+        tasks[numTasks] = t;
+        numTasks += 1;
+    }
+
+    public static void executeEvent(String input) {
+        String[] parsedCommand = parseDeadlineOrEvent(input);
+        String description = parsedCommand[0];
+        String at = parsedCommand[1];
+        Event t = new Event(description, at);
+        tasks[numTasks] = t;
+        numTasks += 1;
+    }
+
+    //Returns a String array with the first element as the description  and the second element as the deadline/time period
+    public static String[] parseDeadlineOrEvent(String input) {
+        String command = getCommand(input);
+        String[] inputArray;
+        if (command.equals("deadline")) {
+            inputArray = input.split(" /by ");
+        } else {
+            inputArray = input.split(" /at ");
+        }
+        inputArray[0] = inputArray[0].substring(inputArray[0].indexOf(" "));
+        return inputArray;
+    }
+
+    public static void listTasks() {
+        if (numTasks == 0) {
+            System.out.println("You don't seem to have any tasks so far!");
+        } else {
+            for (int i = 0; i < numTasks; i++) {
+                System.out.println((i + 1) + ". " + tasks[i]);
+            }
+        }
+    }
+
+    public static String getCommand(String input) {
+        String[] arrayOfInput = input.split(" ", 0);
+        return arrayOfInput[0];
+    }
+
+    public static String getDescription(String input) {
+        return input.substring(input.indexOf(" "));
+    }
+
+    public static String getInput() {
+        System.out.println("Type a valid command (bye, list, mark, unmark, todo, deadline, event) or add a task to your list:");
+        Scanner scan = new Scanner(System.in);
+        return scan.nextLine();
+    }
+
+    public static int getInteger() {
```

Comment:
The naming of some of the methods is misleading such as this. Perhaps getTaskNumber might be more appropriate